### DISCUSSION TITLE
Strict mode, similar to Tailwind's JIT syntax

### DISCUSSION THREAD

**Discussion by innocenzi** [windicss/windicss#181] [Ideas]
Tailwind's JIT only allows arbitrary values by explicitly opting-out of the design system with square brackets (eg. `mt-[1.55rem]`). I think this is very practical: it enforces consistency, while offering an escape gate when needed. 

Could there be a way for Windi to work the same? Perhaps via a configuration option, or by default after a major release?

**Comment by ahku** [comment]
There is also a value with conforming to the official standards,for people who might consider migrating. 

It's one thing if a feature isn't supported at all, like utility groups, but here the official solution kind of solves a lot of things at the same time, like the `.bg-hex-ff0000` or `.mt-2rem` inference. While they do work, they are inconsistent naming-wise. The escape hatch introduced by TW JIT is admittedly more elegant as it both communicates the purpose clearly as well as avoiding various naming problems.

- `.bg-[#f00]`
- `.mb-[2rem]`
- `.bg-image-[https://...]` (will this work...?)

**Comment by antfu** [comment]
Agreed, and I think we can have 

- make auto inferring with brackets works out-of-box
- an option to disabled the auto inferring without the brackets

**Reply by innocenzi** [reply]
That would be nice. 👍

**Comment by ahku** [comment]
@antfu The bracket syntax only works with margins and padding for now? Could not get colors to work, eg. `.bg-[#ff0000]`.

**Reply by antfu** [reply]
https://github.com/windicss/windicss/pull/204

**Reply by ahku** [reply]
What's the best way to update to latest windi/Vite plugin? Can't get the latest fix to work.

**Reply by antfu** [reply]
It haven't been merged/released, likely to do that tmr. Please give a little bit more patient. Thanks

**Comment by ahku** [comment]
@antfu Found some new use cases:

`z-[value]`
`(text/bg/border/etc-)opacity-[value]`

Luckily auto inference seem to be working for opacity, but for z-index I had to resort to inline styling.

**Comment by eminos** [comment]
z-value or z-[value] still doesn't seem to work. Is this not implemented?

[edit]
 nvm, it does work.
